versioning/BRM/masterdbrmnode.cpp

```diff
@@ -171,11 +158,80 @@ void MasterDBRMNode::initMsgQueues(config::Config* config)
 
     serverLock.unlock();
 
-    for (i = 1; i <= NumWorkers; i++)
+    connectToWorkers(connectTimeoutSecs);
+}
+
+void MasterDBRMNode::getNumWorkersAndTimeout(size_t& connectTimeoutSecs,
+                                             const std::string& methodName,
+                                             config::Config* config)
+{
+    string stmp;
+    int ltmp;
+    connectTimeoutSecs = 30; // default
+
+    stmp = config->getConfig("DBRM_Controller", "NumWorkers");
+
+    if (stmp.length() == 0)
+        throw runtime_error(methodName +
+                            ": config file error looking for <DBRM_Controller><NumWorkers>");
+
+    ltmp = static_cast<int>(config::Config::fromText(stmp));
+
+    if (ltmp < 1)
+        throw runtime_error(methodName + ": Bad NumWorkers value");
+
+    NumWorkers = ltmp;
+    stmp = config->getConfig("DBRM_Controller", "WorkerConnectionTimeout");
+    if (stmp.length() > 0)
     {
-        snprintf(ctmp, 50, "DBRM_Worker%d", i);
-        std::string module(ctmp);
-        slaves.push_back(MessageQueueClientPool::getInstance(module));
+        ltmp = static_cast<int>(config::Config::fromText(stmp));
+        if (ltmp > 1)
+            connectTimeoutSecs = ltmp;
+    }
+}
+
+void MasterDBRMNode::connectToWorkers(const size_t connectTimeoutSecs)
+{
+    size_t timeoutMsecs = connectTimeoutSecs * 1000000;
```

Comment:
minor: msecs can be interpreted as milliseconds, but I believe you mean microseconds. a rename of variable can clarify intention